test/unit/test_credentials_holder.py

```diff
@@ -57,9 +74,54 @@ def test_aws_profile_credentials_holder_should_have_session():
 
 def test_aws_profile_credentials_holder_get_session_credentials():
     profile_val: str = "myprofile"
-    obj: AWSProfileCredentialsHolder = AWSProfileCredentialsHolder(profile=profile_val, session=MagicMock())
+    obj: AWSProfileCredentialsHolder = AWSProfileCredentialsHolder(
+        profile=profile_val, session=MagicMock()
+    )
 
     ret_value = obj.get_session_credentials()
     assert len(ret_value) == 1
 
     assert ret_value["profile"] == profile_val
+
+
+@pytest.mark.parametrize(
+    "expiration",
+    [
+        datetime.datetime.now() - datetime.timedelta(hours=3),  # expired 3 hrs ago
+        datetime.datetime.now() - datetime.timedelta(days=1),  # expired 1 day ago
+        datetime.datetime.now() - datetime.timedelta(weeks=1),  # expired 1 week ago
+    ],
+)
+def test_is_expired_true(expiration):
+    credentials: typing.Dict[str, typing.Any] = {
+        "AccessKeyId": "something",
+        "SecretAccessKey": "secret",
+        "SessionToken": "fornow",
+        "Expiration": expiration.astimezone(),
+    }
+
+    obj: CredentialsHolder = CredentialsHolder(credentials=credentials)
+
+    assert obj.is_expired() == True
+
+
+@pytest.mark.parametrize(
+    "expiration",
+    [
+        datetime.datetime.now(),
```

Comment:
Could we use add a minute or so onto this test case (i.e. `datetime.datetime.now() + datetime.timedelta(minutes=1)`)? It's expired by the time the `is_expired()` check occurs on my machine.

Reply:
Thank you, I understand your point. Addressed in 922d7b3